core/templates/dev/head/pages/library/Library.js

```diff
@@ -55,7 +51,17 @@ oppia.controller('Library', [
 
     $scope.activeGroupIndex = null;
 
-    $scope.pageMode = GLOBALS.PAGE_MODE;
+    $scope.pageMode = '';
+    var currentPath = $window.location.pathname;
+    if (currentPath === '/library') {
+      $scope.pageMode = LIBRARY_PAGE_MODES.INDEX;
+    } else if (currentPath === '/library/top_rated') {
+      $scope.pageMode = LIBRARY_PAGE_MODES.GROUP;
+    } else if (currentPath === '/search/find') {
+      $scope.pageMode = LIBRARY_PAGE_MODES.SEARCH;
+    } else {
+      $log.error('INVALID URL PATH: ' + currentPath);
+    }
```

Comment:
Instead of having this block of `if`s you could have data structure in format of 
```javascript
smth = {
  '/search/find': 'search',
  '/library/top_rated': 'group'
}
```
and then do `$scope.pageMode = smth[currentPath]`

Reply:
Done.